### DISCUSSION TITLE
react-query and @loadable/component

### DISCUSSION THREAD

**Discussion by lezan** [TanStack/query#2859] [Q&A]
Does someone know if it possible to use `@loadable/component` with `react-query` instead of Suspense and lazy? I am using GatsbyJS and I am not able to use Suspense and so on, so I switched to `@loadable/component`, but I think `QueryClientProvider` is looking for a Suspense component as children if `suspense: true` in `QueryClient` and it does not recognize a component imported with @loadable/component. If I switch back Suspense and lazy all run smooth in dev environment (not on production of course), so I think I did not something wrong with `react-query`.

Thanks.

**Comment by TkDodo** [comment]
using suspense for data fetching is different from using suspense for code splitting as far as I'm aware.

what we do when `suspense` is enabled is throw the Promise during render so that the nearest Suspense boundary can pick it up, because that's how suspense works. I'm not sure `@loadable/component ` works that way ,too...

**Reply by lezan** [reply]
Hello @TkDodo, thanks for you answer.

I do not have the knowledge to understand how things work in both cases, but code splitting was not my need.

If anyone has any idea how to make it work, I'm here to do all the testing!
